platforms/documentation/docs/src/docs/userguide/dep-man/basics/dependency_configurations.adoc

```diff
@@ -0,0 +1,158 @@
+They determine how and when dependencies are used in various stages of the build process.
+
+Configurations are a fundamental part of dependency resolution in Gradle.
+
+[[sec:what-are-dependency-configurations]]
+== Understanding dependency configurations
+
+Gradle represents the scope of a dependency with the help of a link:{groovyDslPath}/org.gradle.api.artifacts.Configuration.html[Configuration].
+Every configuration can be identified by a unique name.
+
+Many Gradle plugins add pre-defined configurations to your project.
+
+The <<java_plugin.adoc#sec:java_plugin_and_dependency_management,Java plugin>>, which is used for building Java projects and provides a way to compile, package, and test Java applications, adds many configurations.
+These configurations represent the various classpaths needed for source code compilation, executing tests, and more:
+
+[cols="~,~,~"]
+|===
+|Configuration Name |Description |Used to:
+
+|`implementation`
+|Dependencies required for both compilation and runtime.
+|Declare Dependencies
+
+|`compileOnly`
+|Dependencies needed only for compilation, not included in runtime or publication.
+|Declare Dependencies
+
+|`runtimeOnly`
+|Dependencies needed only at runtime, not included in the compile classpath.
+|Declare Dependencies
+
+|`compileClasspath`
+|Combines `implementation` and `compileOnly` dependencies, used for compiling source code.
+|Resolve Dependencies
+
+|`runtimeClasspath`
+|Combines `implementation` and `runtimeOnly` dependencies, used for running the application.
+|Resolve Dependencies
+
+|`annotationProcessor`
+|Dependencies for annotation processors, used during compilation.
+|Resolve Dependencies
+
+|`testImplementation`
+|Dependencies required for compiling and running tests.
+|Declare Dependencies
+
+|`testCompileOnly`
+|Dependencies needed only for test compilation.
+|Declare Dependencies
+
+|`testRuntimeOnly`
+|Dependencies needed only for running tests.
+|Declare Dependencies
+
+|`testCompileClasspath`
+|Combines `testImplementation` and `testCompileOnly` dependencies, used for compiling test sources.
+|Resolve Dependencies
+
+|`testRuntimeClasspath`
+|Combines `testImplementation` and `testRuntimeOnly` dependencies, used for running tests.
+|Resolve Dependencies
+
+|`testAnnotationProcessor`
+|Dependencies for annotation processors used for compiling test sources.
+|Resolve Dependencies
+
+|===
+
+=== Declare Dependencies
+
+The *dependency declaration configurations* (`compileOnly`, `implementation`, `runtimeOnly`) focus on declaring and managing dependencies based on their usage (compile time, runtime, API exposure):
+
+[source,kotlin]
+----
+dependencies {
+    implementation("com.google.guava:guava:30.1.1-jre")     // Implementation dependency
+    compileOnly("org.projectlombok:lombok:1.18.20")         // Compile-only dependency
+    runtimeOnly("mysql:mysql-connector-java:8.0.23")        // Runtime-only dependency
+}
+----
+
+=== Resolve Dependencies
```

Comment:
I'm not sure it makes sense to describe resolvable configurations here. 

We should focus on these in a dedicated page that talks about resolution

Reply:
I do need to introduce these concepts somewhere. I consolidated it to "Others".